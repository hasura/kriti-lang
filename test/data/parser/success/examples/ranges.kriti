# Range tests
[ {{ range i, x := $foo.bar[0] }} {"hello": "world"} {{ end }},
  {{ range _, x := $foo.bar[0] }} {"hello": "world"} {{ end }},
  {{ range _, x := $foo.bar[0] }}
    {"hello": {{$foo.bar }}}
    # Random Comment
  {{ end }},
  [{{ range i, x := {{ if true }} $.events {{ else }} [] {{ end }} }} { "name": {{ x.name }} } {{ end }}],
  [{{ range i, x := $.events }} { "name": {{ x.name }} } {{ end }}]
]