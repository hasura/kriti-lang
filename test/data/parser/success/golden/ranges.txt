Array
    ( Span
        { start = AlexSourcePos
            { line = 1
            , col = 1
            }
        , end = AlexSourcePos
            { line = 8
            , col = 2
            }
        }
    )
    [ Range
        ( Span
            { start = AlexSourcePos
                { line = 1
                , col = 3
                }
            , end = AlexSourcePos
                { line = 1
                , col = 63
                }
            }
        )
        ( Just "i" ) "x"
        ( Path
            ( Span
                { start = AlexSourcePos
                    { line = 1
                    , col = 20
                    }
                , end = AlexSourcePos
                    { line = 1
                    , col = 31
                    }
                }
            )
            [ Obj
                ( Span
                    { start = AlexSourcePos
                        { line = 1
                        , col = 20
                        }
                    , end = AlexSourcePos
                        { line = 1
                        , col = 24
                        }
                    }
                ) NotOptional "$foo" Head
            , Obj
                ( Span
                    { start = AlexSourcePos
                        { line = 1
                        , col = 24
                        }
                    , end = AlexSourcePos
                        { line = 1
                        , col = 28
                        }
                    }
                ) NotOptional "bar" DotAccess
            , Arr
                ( Span
                    { start = AlexSourcePos
                        { line = 1
                        , col = 28
                        }
                    , end = AlexSourcePos
                        { line = 1
                        , col = 31
                        }
                    }
                ) NotOptional 0
            ]
        )
        ( Object
            ( Span
                { start = AlexSourcePos
                    { line = 1
                    , col = 35
                    }
                , end = AlexSourcePos
                    { line = 1
                    , col = 53
                    }
                }
            )
            ( fromList
                [
                    ( "hello"
                    , StringTem
                        ( Span
                            { start = AlexSourcePos
                                { line = 1
                                , col = 45
                                }
                            , end = AlexSourcePos
                                { line = 1
                                , col = 52
                                }
                            }
                        )
                        [ String
                            ( Span
                                { start = AlexSourcePos
                                    { line = 1
                                    , col = 46
                                    }
                                , end = AlexSourcePos
                                    { line = 1
                                    , col = 51
                                    }
                                }
                            ) "world"
                        ]
                    )
                ]
            )
        )
    , Range
        ( Span
            { start = AlexSourcePos
                { line = 2
                , col = 3
                }
            , end = AlexSourcePos
                { line = 2
                , col = 63
                }
            }
        ) Nothing "x"
        ( Path
            ( Span
                { start = AlexSourcePos
                    { line = 2
                    , col = 20
                    }
                , end = AlexSourcePos
                    { line = 2
                    , col = 31
                    }
                }
            )
            [ Obj
                ( Span
                    { start = AlexSourcePos
                        { line = 2
                        , col = 20
                        }
                    , end = AlexSourcePos
                        { line = 2
                        , col = 24
                        }
                    }
                ) NotOptional "$foo" Head
            , Obj
                ( Span
                    { start = AlexSourcePos
                        { line = 2
                        , col = 24
                        }
                    , end = AlexSourcePos
                        { line = 2
                        , col = 28
                        }
                    }
                ) NotOptional "bar" DotAccess
            , Arr
                ( Span
                    { start = AlexSourcePos
                        { line = 2
                        , col = 28
                        }
                    , end = AlexSourcePos
                        { line = 2
                        , col = 31
                        }
                    }
                ) NotOptional 0
            ]
        )
        ( Object
            ( Span
                { start = AlexSourcePos
                    { line = 2
                    , col = 35
                    }
                , end = AlexSourcePos
                    { line = 2
                    , col = 53
                    }
                }
            )
            ( fromList
                [
                    ( "hello"
                    , StringTem
                        ( Span
                            { start = AlexSourcePos
                                { line = 2
                                , col = 45
                                }
                            , end = AlexSourcePos
                                { line = 2
                                , col = 52
                                }
                            }
                        )
                        [ String
                            ( Span
                                { start = AlexSourcePos
                                    { line = 2
                                    , col = 46
                                    }
                                , end = AlexSourcePos
                                    { line = 2
                                    , col = 51
                                    }
                                }
                            ) "world"
                        ]
                    )
                ]
            )
        )
    , Range
        ( Span
            { start = AlexSourcePos
                { line = 3
                , col = 3
                }
            , end = AlexSourcePos
                { line = 6
                , col = 12
                }
            }
        ) Nothing "x"
        ( Path
            ( Span
                { start = AlexSourcePos
                    { line = 3
                    , col = 20
                    }
                , end = AlexSourcePos
                    { line = 3
                    , col = 31
                    }
                }
            )
            [ Obj
                ( Span
                    { start = AlexSourcePos
                        { line = 3
                        , col = 20
                        }
                    , end = AlexSourcePos
                        { line = 3
                        , col = 24
                        }
                    }
                ) NotOptional "$foo" Head
            , Obj
                ( Span
                    { start = AlexSourcePos
                        { line = 3
                        , col = 24
                        }
                    , end = AlexSourcePos
                        { line = 3
                        , col = 28
                        }
                    }
                ) NotOptional "bar" DotAccess
            , Arr
                ( Span
                    { start = AlexSourcePos
                        { line = 3
                        , col = 28
                        }
                    , end = AlexSourcePos
                        { line = 3
                        , col = 31
                        }
                    }
                ) NotOptional 0
            ]
        )
        ( Object
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 5
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 29
                    }
                }
            )
            ( fromList
                [
                    ( "hello"
                    , Path
                        ( Span
                            { start = AlexSourcePos
                                { line = 4
                                , col = 17
                                }
                            , end = AlexSourcePos
                                { line = 4
                                , col = 25
                                }
                            }
                        )
                        [ Obj
                            ( Span
                                { start = AlexSourcePos
                                    { line = 4
                                    , col = 17
                                    }
                                , end = AlexSourcePos
                                    { line = 4
                                    , col = 21
                                    }
                                }
                            ) NotOptional "$foo" Head
                        , Obj
                            ( Span
                                { start = AlexSourcePos
                                    { line = 4
                                    , col = 21
                                    }
                                , end = AlexSourcePos
                                    { line = 4
                                    , col = 25
                                    }
                                }
                            ) NotOptional "bar" DotAccess
                        ]
                    )
                ]
            )
        )
    , Array
        ( Span
            { start = AlexSourcePos
                { line = 7
                , col = 3
                }
            , end = AlexSourcePos
                { line = 7
                , col = 68
                }
            }
        )
        [ Range
            ( Span
                { start = AlexSourcePos
                    { line = 7
                    , col = 4
                    }
                , end = AlexSourcePos
                    { line = 7
                    , col = 67
                    }
                }
            )
            ( Just "i" ) "x"
            ( Path
                ( Span
                    { start = AlexSourcePos
                        { line = 7
                        , col = 21
                        }
                    , end = AlexSourcePos
                        { line = 7
                        , col = 29
                        }
                    }
                )
                [ Obj
                    ( Span
                        { start = AlexSourcePos
                            { line = 7
                            , col = 21
                            }
                        , end = AlexSourcePos
                            { line = 7
                            , col = 22
                            }
                        }
                    ) NotOptional "$" Head
                , Obj
                    ( Span
                        { start = AlexSourcePos
                            { line = 7
                            , col = 22
                            }
                        , end = AlexSourcePos
                            { line = 7
                            , col = 29
                            }
                        }
                    ) NotOptional "events" DotAccess
                ]
            )
            ( Object
                ( Span
                    { start = AlexSourcePos
                        { line = 7
                        , col = 33
                        }
                    , end = AlexSourcePos
                        { line = 7
                        , col = 57
                        }
                    }
                )
                ( fromList
                    [
                        ( "name"
                        , Path
                            ( Span
                                { start = AlexSourcePos
                                    { line = 7
                                    , col = 46
                                    }
                                , end = AlexSourcePos
                                    { line = 7
                                    , col = 52
                                    }
                                }
                            )
                            [ Obj
                                ( Span
                                    { start = AlexSourcePos
                                        { line = 7
                                        , col = 46
                                        }
                                    , end = AlexSourcePos
                                        { line = 7
                                        , col = 47
                                        }
                                    }
                                ) NotOptional "x" Head
                            , Obj
                                ( Span
                                    { start = AlexSourcePos
                                        { line = 7
                                        , col = 47
                                        }
                                    , end = AlexSourcePos
                                        { line = 7
                                        , col = 52
                                        }
                                    }
                                ) NotOptional "name" DotAccess
                            ]
                        )
                    ]
                )
            )
        ]
    ]